logtail/logtail.go

```diff
@@ -757,6 +767,33 @@ func (l *Logger) Write(buf []byte) (int, error) {
 	return inLen, err
 }
 
+func redactIPs(buf []byte) []byte {
+	regexMatchesIPv6 := regexp.MustCompile(`([0-9a-fA-F]{1,4}):([0-9a-fA-F]{1,4}):([0-9a-fA-F:]{1,4})*`)
+	regexMatchesIPv4 := regexp.MustCompile(`(\d{1,3})\.(\d{1,3})\.\d{1,3}\.\d{1,3}`)
+
+	out := regexMatchesIPv6.ReplaceAllStringFunc(string(buf), func(s string) string {
```

Comment:
I would guess we can save some allocation and copying if we use the byte forms:

```go
regexMatchesIPv6.ReplaceAllFunc(buf, func(b []byte) []byte {
   // ...
   prefix := bytes.Split(b, []byte(":"))
   return bytes.Join(append(prefix[:2], []byte("x")), []byte(":"))
})
```

That way we don't need to convert to string and back, since we wanted the output as a byte slice anyway. But, YMMV.